prov/sockets/src/sock_epoll.c

```diff
@@ -197,3 +197,114 @@ void sock_epoll_close(struct sock_epoll_set *set)
 }
 
 #endif
+
+int sock_epoll_map_init(struct sock_epoll_map *emap)
+{
+	memset(emap, 0, sizeof(*emap));
+
+	fastlock_init(&emap->lock);
+
+	if (sock_epoll_create(&emap->set, sock_cm_def_map_sz) < 0) {
+		SOCK_LOG_ERROR("failed to create epoll set: %s\n", strerror(errno));
+		return -FI_ENOMEM;
+	}
+
+	if (socketpair(AF_UNIX, SOCK_STREAM, 0, emap->signal_fds) < 0)
```

Comment:
Such a call to `socketpair()` is already present in Libfabric's master code, without distinction if the code is running on Linux, FreeBSD, OS X or Windows (see function `sock_conn_listen()` in `sock_conn.c`)